### PR TITLE
STYLE: Use MakeFilled, () or {}, instead of `Vector(const ValueType &)`

### PR DESCRIPTION
**Pull request by N-Dekker**
A call to `MakeFilled<VectorType>(value)` expresses more clearly than
`Vector(const ValueType &)` that an _entire_ Vector is to be filled by
the specific scalar value.

Zero-initialization of a Vector is achieved in the most generic way, by
an empty pair of parentheses `()` or an empty pair of braces `{}`,
rather than by passing 0.0 to the explicit `Vector(const ValueType &)`
constructor.

### CODE REVIEW

**Comment by N-Dekker** [comment]
> Do you have a script to do this, or a procedure to do this? All the Remote modules will need to be updated as well.

@hjmjohnson  This is just a style PR. And it's a draft anyway! Only _implicit_ conversion from a scalar to a Vector is now deprecated (legacy-only). But I think it's clearer to just replace _explicit_ Vector(scalar) constructions as well. When possible, and when it's obvious how to do it!

I did so manually, by locally deleting the `Vector(const ValueType &)` and then addressing the local errors. (It is not my intention to delete `Vector(const ValueType &)` from ITK. Certainly not from ITK 5.3!  😃 Probably not even from ITK 5.x. )

File: Modules/Filtering/DisplacementField/include/itkComposeDisplacementFieldsImageFilter.hxx [hunk 1]
```diff
@@ -60,7 +60,7 @@ template <typename InputImage, typename TOutputImage>
 void
 ComposeDisplacementFieldsImageFilter<InputImage, TOutputImage>::BeforeThreadedGenerateData()
 {
-  VectorType zeroVector(0.0);
+  VectorType zeroVector{};
```
Comment (Leengit) [inline Modules/Filtering/DisplacementField/include/itkComposeDisplacementFieldsImageFilter.hxx:63]: I bet many of these `zeroVector` variables could be `const`.  (Perhaps for another day.)

**Review by dzenanz — APPROVED** [review]
LGTM

File: Modules/Core/Common/include/itkNumericTraitsVectorPixel.h [hunk 2]
```diff
@@ -72,43 +72,43 @@ class NumericTraits<Vector<T, D>>
...
   {
-    return Self(NumericTraits<T>::NonpositiveMin());
+    return MakeFilled<Self>(NumericTraits<T>::NonpositiveMin());
   }
 
   static const Self
   ZeroValue()
   {
-    return Self(NumericTraits<T>::ZeroValue());
+    return MakeFilled<Self>(NumericTraits<T>::ZeroValue());
```
[the remark this answers is not in the document]
Comment (N-Dekker) [reply Modules/Core/Common/include/itkNumericTraitsVectorPixel.h:105]: @Leengit I think you're right, but are you sure that `NumericTraits<T>::ZeroValue()` is always equal to `T{}`. If so, we might as well just do:

    return {};

File: Modules/Filtering/DisplacementField/include/itkComposeDisplacementFieldsImageFilter.hxx (hunk 1 above)
Comment (N-Dekker) [reply Modules/Filtering/DisplacementField/include/itkComposeDisplacementFieldsImageFilter.hxx:63]: I bet many of these `zeroVector` variables could even be `constexpr`  😸 I still hesitate about including such changes with this commit. Maybe we could run a tool to constify (or "constexprify") local variables automatically, independent of this PR....?

Comment (hjmjohnson) [reply Modules/Filtering/DisplacementField/include/itkComposeDisplacementFieldsImageFilter.hxx:63]: @N-Dekker  Let me know if you find such a "constify" tool.  I have a few hundred projects I'd love to run it on :).

**Review by hjmjohnson — APPROVED** [review]

File: Modules/Core/Common/include/itkNumericTraitsVectorPixel.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (N-Dekker) [reply Modules/Core/Common/include/itkNumericTraitsVectorPixel.h:105]: @Leengit I would very much like if we could reduce the use of `NumericTraits<T>::ZeroValue()` in favor of built-in `0`, `{}`, or `T()`, whenever they would mean the same thing, but I think it's beyond the scope of this pull request. Thanks for the suggestion anyway!

File: Modules/Core/Common/test/itkImageBaseGTest.cxx [hunk 3]
```diff
@@ -130,8 +130,8 @@ CheckInvalidSpacingExceptions()
 
   // Test exceptions
   const SpacingType initialSpacing = imageBase->GetSpacing();
-  const SpacingType negativeSpacing(-1.0);
-  const SpacingType zeroSpacing(0.0);
+  const auto        negativeSpacing = itk::MakeFilled<SpacingType>(-1.0);
```
[the remark this answers is not in the document]
Comment (N-Dekker) [reply Modules/Core/Common/test/itkImageBaseGTest.cxx:133]: @Leengit  I understand your point, but I would like the reader to be able to brainlessly assume that a variable initialized by `MakeFilled<SpecificContainerType>` is of that specific `SpecificContainerType` type 😺

**Comment by N-Dekker** [comment]
This [force-push](https://github.com/InsightSoftwareConsortium/ITK/compare/41a4ad06592a856102867b6129d4862abc949ed7..478eb431c24338d72f1c53453db2213afebeabda) is intended to be the only large addition of changes to this PR... I promise 😃

**Review by hjmjohnson — APPROVED** [review]

**Comment by hjmjohnson** [comment]
@N-Dekker This is still a draft.

**Comment by N-Dekker** [comment]
> This is still a draft.

@hjmjohnson  Just marked ready, which means that I have approved the PR myself  😸

**Comment by N-Dekker** [comment]
Thanks for merging @hjmjohnson and for your approval @dzenanz 

Please note that this PR has not entirely removed each and every use of `Vector(const ValueType &)`. It just did the "low hanging fruit". The cases that are still there appear less obvious to me. You may find them by locally adding `[[deprecated]]` to the  `Vector(const ValueType &)` constructor(s), and then checking the warnings.

Again, it isn't my intention to entirely _ban_ the  `Vector(const ValueType &)` constructor. Certainly not for ITK 5.3! But it often appears clearer to either call `MakeFilled<VectorType>(value)`, or (in case of zero-initialization) use the `VectorType()` or `{}` syntax.

**Pull request merged**